### ISSUE TITLE
v2 nested facet cursor get error

### ISSUE THREAD

**Issue by marbemac** [Yomguithereal/baobab#309]
Hey @Yomguithereal, say I have a structure like this:

```js
const defaultData = {
  models: {
    projects: {},
  },
  currentProject: {
    id: null,
    $data: {
      cursors: {
        id: ['currentProject', 'id'],
        projects: ['models', 'projects']
      },
      get(data) {
        return data.projects[data.id] || {}
      }
    }
  },
}
```

If I try to get the currentProject (state.get('currentProject')) part of the tree, I get a:

```
Can't add property $data, object is not extensible
```

I can access the individual elements just fine though, ie state.get(['currentProject', 'id']) and state.get(['currentProject', '$data']) work as expected.

Is this the expected behavior?

**Comment by marbemac** [comment]
Also, this error means I can't subscribe to ['currentProject'] in my react components with baobab-react.

**Comment by Yomguithereal** [comment]
Hello @marbemac. I cannot reproduce your bug. Can you tell me on which dev version you are and if you passed specific options to your tree, please?

**Comment by marbemac** [comment]
@Yomguithereal strange, I'm on dev11, with default options.

I just noticed I have to set something inside the part of the tree w the facet before the get. Try the following:

state.set(['currentProject', 'loading'], true)
state.get('currentProject')

Does that error for you? If not, I'll go ahead and setup a dedicated repo to isolate the issue.

**Comment by Yomguithereal** [comment]
Ok, this should be solved for `2.0.0-dev12`.

**Comment by marbemac** [comment]
You're the man - working! Thanks.
